### PR TITLE
Merchants no longer sell empty cannabis and other drug plants

### LINKED ISSUE #5843: Special drugs purchased from traders do not contain proper reagents
Submitter: airconswitch
**Describe the bug**
Purchasing things like omega weed or white weed from a random event vendor results in a leaf of that description, minus the associated reagents.

**To Reproduce**
1. Vendor random event happens where they offer stuff like omega weed
2. Buy omega weed
3. Feel ripped off

**Expected behavior**
Weed should have proper reagents

**Screenshots**
None

**Additional context**
None

Reported on: Goonstation 3 Roleplay: Morty 2021-08-30 02:05:01

### PR DESCRIPTION
**Pull request by Nexusuxen**
[BUG]

## About the PR 

Fixes #5843 by changing the commodity atoms for weed, its variants, poppies, and magic mushrooms to use their spawnable versions. Poppies and magic mushrooms now also have spawnable versions, which are now used here. They default to 15u of morphine and psilocybin respectively.

## Why's this needed? 

I would prefer for my hard earned credits to buy not-empty magic mushrooms, thank you very much.

## Changelog

```changelog
(u)Nexusuxen
(+)Cannabis, magic mushrooms, and poppies bought from traders now actually contain drugs.
```

### CODE REVIEW

File: code/datums/commodity.dm [hunk 1]
```diff
@@ -659,47 +659,47 @@
 
 /datum/commodity/drugs/poppies
 	comname = "Poppies"
-	comtype = /obj/item/plant/herb/poppy
```
Comment (ZeWaka) [inline code/datums/commodity.dm:662]: doesn't this prevent them from buying them, however?

Comment (Nexusuxen) [reply code/datums/commodity.dm:662]: I'd imagine `comtype = /obj/item/plant/herb/poppy/spawnable`, the line replacing the highlighted line, should work? I had difficulties getting RNG to comply during tests but for the cannabis and mushrooms I had the expected behavior.

Comment (ZeWaka) [reply code/datums/commodity.dm:662]: doesn't that make it so they can only buy the spawnable variety?

Comment (Nexusuxen) [reply code/datums/commodity.dm:662]: Correct, the spawnable variety just comes with morphine, whereas the regular version doesn't. From my understanding this is consistent with the rest of the changes I made; I assume I overlooked something?

Comment (ZeWaka) [reply code/datums/commodity.dm:662]: I mean the players selling to the trader. They will only be able to sell the spawnable variant to them, of which they cannot farm themselves.

Comment (Nexusuxen) [reply code/datums/commodity.dm:662]: OH, fuck okay, good catch, thank you!! I'll fix that soon; just need to make a new spawnable variant of the drugs and slap those into the sold versions.

**Comment by Nexusuxen** [comment]
Running into issues with how weirdly the spawnable items are and trader code, so this'll be a bit to actually get to working order; that and my ADHD is going brrrrr today so it's incredibly hard to not get distracted for the moment.

**Comment by Nexusuxen** [comment]
I've been putting off continuing this PR simply due to the fact that fixing this would require adding onto a poorly done spawnable system (the spawnable drugs should come from a spawner obj and not be a child of the drug obj itself), OR refactoring it and fixing all the issues that come from how many references to spawnable weed n shit there is, which is *way* more than I anticipated I'd be doing to fix this issue. 

For the time being I'm just gonna close this, but I might revisit it in the future if nobody else does.

**Pull request closed without merging**